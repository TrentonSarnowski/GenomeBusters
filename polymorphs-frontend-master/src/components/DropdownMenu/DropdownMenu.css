.DropdownMenu-label {
}

.DropdownMenu-select-menu {
  color: var(--lightest-color);
  background-color: var(--darkest-color);
  text-align: left;
  position: relative;
  top: 15px;
  left: -175px;
  width: 260px;
}

.DropdownMenu-select-menu img {
  position: absolute;
  right: 0px;
  top: -15px;
}

.DropdownMenu-select-menu ul {
  margin-top: 5px;
  list-style-type: none;
  padding-left: 0px;
}

.DropdownMenu-select-menu ul li {
  padding-left: 20px;
  cursor: default;
}
.DropdownMenu-select-menu ul li:hover{
  background-color: #3f3f3f;
}

.DropdownMenu-select-menu.show {
  visibility: show;
}

.DropdownMenu-select-menu.hidden {
  visibility: hidden;
}
