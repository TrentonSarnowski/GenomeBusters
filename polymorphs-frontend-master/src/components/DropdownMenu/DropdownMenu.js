import React, { Component } from 'react';
import './DropdownMenu.css';

class DropdownMenu extends Component {
  constructor(props) {
    super(props);


    this.state = {
      active: false,
    };

    this.toggle = this.toggle.bind(this);
    this.select = this.select.bind(this);
  }

  toggle() {
    this.setState({active: !this.state.active});
  }

  select(id) {
    this.props.fetch_preset_function(id);
    this.setState({active: false});
  }

  render() {
    return (
      <div className="App-cell App-select">
        <div className="DropdownMenu-label" onClick={this.toggle}>Select</div>
        <div className={`DropdownMenu-select-menu ${this.state.active ? 'show' : 'hidden'}`}>
          <img role="presentation" src="/triangle.svg" />
          <ul>
            {this.props.presets.map(function(genome) {
              return <li onClick={() => this.select(genome.id)} key={genome.id}>{genome.name}</li>
            }, this)}
          </ul> 
        </div>
      </div>
    );
  }
}

export default DropdownMenu;
