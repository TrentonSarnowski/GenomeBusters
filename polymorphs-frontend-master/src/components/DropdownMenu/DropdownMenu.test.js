import React from 'react';
import ReactDOM from 'react-dom';
import DropdownMenu from './DropdownMenu';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const tree = renderer.create(<DropdownMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});

