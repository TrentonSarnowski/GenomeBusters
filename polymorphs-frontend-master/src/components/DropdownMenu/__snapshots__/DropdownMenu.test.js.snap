exports[`test renders without crashing 1`] = `
<div
  className="App-cell App-select">
  <div
    className="DropdownMenu-label"
    onClick={[Function]}>
    Select
  </div>
  <div
    className="DropdownMenu-select-menu hidden">
    <img
      role="presentation"
      src="/triangle.svg" />
    <ul>
      <li
        onClick={[Function]}>
        test 1
      </li>
      <li
        onClick={[Function]}>
        test 2
      </li>
    </ul>
  </div>
</div>
`;
